In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
df = pd.read_csv('sampled_data_with_IMDB_Score.csv')

In [5]:
#Setting the value for X and Y
x = df[['tomatometer_rating', 'audience_rating']]
y = df['IMDB_score']

In [8]:
#Splitting the dataset
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 100)

In [10]:
#Fitting the Multiple Linear Regression model
from sklearn.linear_model import LinearRegression
mlr = LinearRegression()  
mlr.fit(x_train, y_train)

LinearRegression()

In [11]:
#Intercept and Coefficient
print("Intercept: ", mlr.intercept_)
print("Coefficients:")
list(zip(x, mlr.coef_))

Intercept:  4.000201569637845
Coefficients:


[('tomatometer_rating', 0.009650805390047645),
 ('audience_rating', 0.02868513837624001)]

Regression Equation: IMDB_score = 4.0002015 + (0.00965080 * tomatometer_rating) + (0.028685* audience_rating)+ e

Prediction on the test set

In [12]:
#Prediction of test set
y_pred_mlr= mlr.predict(x_test)
#Predicted values
print("Prediction for test set: {}".format(y_pred_mlr))

Prediction for test set: [6.31348361 5.47546721 6.21056089 5.96846012 5.43202418 5.77868015
 5.7767804  5.96926195 5.02428486 6.46926169 5.38834268 5.23122821
 4.83501064 5.9092189  6.49444341 5.56930249 6.175165   6.7376133
 5.9783782  6.03684639]


In [14]:
slr_diff = pd.DataFrame({'Actual value': y_test, 'Predicted value': y_pred_mlr})
slr_diff.head()

Actual value  Predicted value
37           6.5         6.313484
62           6.2         5.475467
26           5.0         6.210561
41           5.3         5.968460
35           6.0         5.432024

In [15]:
#Model Evaluation
from sklearn import metrics
meanAbErr = metrics.mean_absolute_error(y_test, y_pred_mlr)
meanSqErr = metrics.mean_squared_error(y_test, y_pred_mlr)
rootMeanSqErr = np.sqrt(metrics.mean_squared_error(y_test, y_pred_mlr))
print('R squared: {:.2f}'.format(mlr.score(x,y)*100))
print('Mean Absolute Error:', meanAbErr)
print('Mean Square Error:', meanSqErr)
print('Root Mean Square Error:', rootMeanSqErr)

R squared: 41.50
Mean Absolute Error: 0.6817817926112723
Mean Square Error: 0.722708806283965
Root Mean Square Error: 0.8501228183527161
